Enable/View Controllers/ColorViewController.m

```diff
@@ -36,19 +88,102 @@ - (NSInteger)pickerView:(nonnull UIPickerView *)pickerView numberOfRowsInCompone
+        [self.labelColorWell setSelectedColor: [singleton getLabelColor]];
+        [self.likeColorWell setSelectedColor: [singleton getLikeColor]];
+        [self.starColorWell setSelectedColor: [singleton getStarColor]];
+    }
+}
 - (void) setupTheme {
     [self setupMainTheme];
-    [self.titleLabel setTextColor:[UIColor colorNamed: [ThemeTracker sharedTheme].colorSet[@"Label"]]];
-    [self.themePicker setBackgroundColor:[UIColor colorNamed: [ThemeTracker sharedTheme].colorSet[@"Secondary"]]];
+    ThemeTracker * singleton = [ThemeTracker sharedTheme];
+    [self.scrollContentView setBackgroundColor: [singleton getBackgroundColor]];
+    [self.customizeContentView setBackgroundColor: [singleton getBackgroundColor]];
+    [self.customizeButton setTintColor: [singleton getAccentColor]];
+    [self.titleLabel setTextColor: [singleton getLabelColor]];
+    [self.backgroundColorLabel setTextColor: [singleton getLabelColor]];
+    [self.secondaryColorLabel setTextColor: [singleton getLabelColor]];
+    [self.accentColorLabel setTextColor: [singleton getLabelColor]];
+    [self.likeColorLabel setTextColor: [singleton getLabelColor]];
+    [self.starColorLabel setTextColor: [singleton getLabelColor]];
+    [self.labelColorLabel setTextColor: [singleton getLabelColor]];
+    [self.themePicker setBackgroundColor: [singleton getSecondaryColor]];
     [self.themePicker reloadComponent:0];
+    [self updateColorWells];
+
+}
+- (IBAction)didTapCustomize:(id)sender {
+    if([self checkColors]){
+        NSDictionary * dict = @{@"Background" : self.backgroundColorWell.selectedColor, @"Secondary" : self.secondaryColorWell.selectedColor,
+                                @"Label" : self.labelColorWell.selectedColor, @"Accent" : self.accentColorWell.selectedColor,
+                                @"Like" : self.likeColorWell.selectedColor, @"Star" : self.starColorWell.selectedColor, @"StatusBar" : [self calculateStatusBar]};
+        [[ThemeTracker sharedTheme] updateTheme:@"Custom" withColorDict:dict];
+    }
+}
+- (NSString *) calculateStatusBar {
+    // taken from https://www.w3.org/WAI/ER/WD-AERT/#color-contrast
+    NSString * retVal = @"Dark";
+    const CGFloat * components = CGColorGetComponents(self.backgroundColorWell.selectedColor.CGColor);
+    float contrastVal = ((components[0] * 255 * 299) + (components[1] * 255 * 587) + (components[2] * 255 * 114)) / 1000;
+    if(contrastVal < 125) {
+        retVal = @"Light";
+    }
+    return retVal;
+}
+
+- (bool) calculateDifferenceColor: (CGColorRef) c1 and: (CGColorRef) c2 {
+    // taken from https://www.w3.org/WAI/ER/WD-AERT/#color-contrast
+    const CGFloat * c1Components = CGColorGetComponents(c1);
+    const CGFloat * c2Components = CGColorGetComponents(c2);
+    float contrast = (fabs(c1Components[0] - c2Components[0]) + fabs(c1Components[1] - c2Components[1]) + fabs(c1Components[2] - c2Components[2])) * 255;
+    return contrast > 200;
```

Comment:
200 should be a constant, as this is something you would probably play around with time